src/ingredients/ingredients.service.ts

```diff
@@ -0,0 +1,83 @@
+import { CreateIngredientDto } from '@/ingredients/dto/create.ingredient.dto';
+import { UpdateIngredientDto } from '@/ingredients/dto/update.ingredient.dto';
+import { IngredientSchema } from '@/schema/ingredients.schema';
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { ReturnModelType } from '@typegoose/typegoose';
+import { Types } from 'mongoose';
+import { InjectModel } from 'nest-typegoose';
+
+@Injectable()
+export class IngredientsService {
+  constructor(
+    @InjectModel(IngredientSchema)
+    private ingredientModel: ReturnModelType<typeof IngredientSchema>,
+  ) {}
+
+  async createIngredient(ingredientInfo: CreateIngredientDto) {
+    try {
+      const createIng = await this.ingredientModel.create({
+        title: ingredientInfo.title,
+        available: ingredientInfo.available,
+      });
+      return createIng.toObject();
```

Comment:
Is it really necessary to call `toObject` right here?

Reply:
NoT^T , Edited